### PR TITLE
ESSNTL-4821 inventory groups

### PR DESCRIPTION
**Pull request by petrblaho**
- Adds assertion to permission test
- Mocks fetch_url response instead of get_perms in RBAC test
- Adds parser for RBAC's inventory group ids

## Secure Coding Practices Checklist Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [x] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [x] General Coding Practices

### CODE REVIEW

**Review by fellipeh — APPROVED** [review]
lgtm

File: kerlescan/rbac_service_interface.py [hunk 1]
```diff
@@ -18,3 +18,76 @@ def get_perms(application, service_auth_key, logger, request_metric, exception_m
...
+            return value
+        elif isinstance(value, str):
+            # split by comma, parses "null" as None
+            return [None if val == "null" else val for val in value.split(",")]
+        else:
+            raise TypeError
+
+    # get only relevant permission records
+    definitions = [
+        perm["resourceDefinitions"] for perm in rbac_data if perm["permission"] in permission_list
```
Comment (vkrizan) [inline kerlescan/rbac_service_interface.py:66]: Lack of `resourceDefinitions` not handled.

File: kerlescan/rbac_service_interface.py [hunk 2]
```diff
@@ -18,3 +18,76 @@ def get_perms(application, service_auth_key, logger, request_metric, exception_m
...
+        else:
+            # get attributeFilters relevant for inventory groups
+            attribute_filters = [
+                definition.get("attributeFilter", {})
+                for definition in definitions
+                if definition.get("attributeFilter", {}).get("key") == "group.id"
+            ]
+
+            if attribute_filters == []:
+                group_ids = None
```
Comment (vkrizan) [inline kerlescan/rbac_service_interface.py:90]: This should be returned if at least one of the permissions does not have an attribute filter, or has attribute filters that are not relevant.

File: kerlescan/rbac_service_interface.py [hunk 3]
```diff
@@ -18,3 +18,76 @@ def get_perms(application, service_auth_key, logger, request_metric, exception_m
...
+            else:
+                group_ids = [
+                    get_attfibute_filter_value(attribute_filter)
+                    for attribute_filter in attribute_filters
+                ]
+                # flatten them
+                if None in group_ids:
+                    group_ids = None
```
Comment (vkrizan) [inline kerlescan/rbac_service_interface.py:98]: Do I understand it correctly that if a None is found in the collected filters, we'd allow all groups?

File: kerlescan/rbac_service_interface.py (hunk 1 above)
Comment (petrblaho) [reply kerlescan/rbac_service_interface.py:66]: https://github.com/RedHatInsights/kerlescan/pull/96

File: kerlescan/rbac_service_interface.py (hunk 3 above)
Comment (petrblaho) [reply kerlescan/rbac_service_interface.py:98]: there is a comment on a wrong line and it is confusing

**Pull request merged**